scripts/travis/build-all-in-one-image.sh

```diff
@@ -28,7 +28,7 @@ run_integration_test() {
 
 upload_to_docker() {
   # Only push the docker container to Docker Hub for master branch
-  if [[ ("$BRANCH" == "master" || $BRANCH =~ ^v[0-9]+\.[0-9]+\.[0-9]+$) && "$TRAVIS_SECURE_ENV_VARS" == "true" ]]; then
```

Comment:
The reason we had this TRAVIS_SECURE_ENV_VARS check was to prevent attempt to deploy snapshot images from pull requests. What would be the equivalent of that?  I assume the DockerHub Token secret won't be available in forks.

PS. It would be good to document this part of the publishing mechanism in RELEASE.md

Reply:
I see. sure.

Reply:
In the logs the value of BRANCH seems to be `merge`: https://github.com/jaegertracing/jaeger/runs/1467144626?check_suite_focus=true#step:8:1639
```
skip docker upload for PR
+ [[ merge == \m\a\s\t\e\r ]]
+ [[ merge =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]
+ echo 'skip docker upload for PR'
```

Reply:
This is fixed.